tests/courses.js

```diff
@@ -0,0 +1,94 @@
+const request = require("supertest");
+const app = require("../server");
+const User = require("../models/User");
+const Course = require("../models/CourseModel");
+const jwtDecode = require('jwt-decode');
+
+var token;
+var userId;
+
+var testUserIds;
+var testStudentId = "aeeeded"
+
+beforeAll(async () => {
+  await request(app)
+    .post("/api/users/register")
+    .send({ 
+      name: "testingUser",
+      email: "user@test.com",
+      password: "testingUser",
+      password2: "testingUser"
+    })
+    .then(response => {
+      expect(response.statusCode).toBe(200);
+    });
+  await request(app)
+    .post("/api/users/login")
+    .send({ 
+      email: "user@test.com",
+      password: "testingUser"
+    })
+    .then(response => {
+      expect(response.statusCode).toBe(200);
+      token = response.body.token
+      userId = jwtDecode(token).id;
+    });
+});
+describe("Test functionality", () => {
+    let courseId;
+    test("Create Course", () => {
+        return request(app)
+        .post("/api/courses/")
```

Comment:
Need to pass jwt as the Authorization header. 

adding `.set({'Authorization': token})` should work.